File: Day5/First.lua
-- Advent of Code 2018. Day 5, first part.
-- Luis Delgado.

require "Process"

Data_File="Data.txt";
local File = io.open (Data_File,"r");

String = File:read()
StringSize = String:len();
local Sz = StringSize -1;

Mem = {};

for i = 0,Sz do
	table.insert(Mem, string.byte(String,i+1));
end

print("Processing. It can take a long time")

PolymerSize = Process(Mem,StringSize)

print("*****");
for k,v in pairs(Mem) do
	io.write(string.char(v))
end
print()
print("(" ..PolymerSize.. ")")

File: Day9/First.py
#   Advent of Code 2018. Day 9, first part.
#   Luis Delgado.

File = open("Data.txt",'r')

import sys
sys.dont_write_bytecode = True

from Import import Extract,Circle

data = File.readline()
Players, Worth = Extract(data)
z

MarbleGame = Circle()
Elves_Points = []
for i in range(Players):
    Elves_Points.append(0)

a=0
for i in range(Worth):
    
    Elves_Points[a]=Elves_Points[a]+MarbleGame.add()
    a=a+1
    if(a==Players):
        a=0

print(max(Elves_Points))

